### ISSUE TITLE
Add rule to enforce custom properties be specified at the top of the scope

### ISSUE THREAD

**Issue by sindresorhus** [stylelint/stylelint#3198]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

I would like to enforce that custom properties be specified at the top of the current scope, for readability reasons.

Fail:

```css
.foo {
  color: pink;
  --margin: 10px;
  margin: var(--margin);
}
```

Pass:

```css
.foo {
  --margin: 10px;
  color: pink;
  margin: var(--margin);
}
```

I would also like an option to require a newline after the group of custom properties:

```css
.foo {
  --margin: 10px;
  --color: pink;

  color: var(--color);
  margin: var(--margin);
}
```

**Comment by hudochenkov** [comment]
For order you can use [`stylelint-order`](https://github.com/hudochenkov/stylelint-order) plugin. Rule [`order/order`](https://github.com/hudochenkov/stylelint-order/blob/master/rules/order/README.md).

There is no way to specify newline _after_ custom properties, but it's possible to specify it _before_ different things. I think you'll want to specify [empty line before declarations](https://stylelint.io/user-guide/rules/declaration-empty-line-before/) (properties).

I believe the following configuration is what you want:

```json
{
	"plugins": ["stylelint-order"],
	"rules": {
		"order/order": ["custom-properties", "declarations"],
		"declaration-empty-line-before": [
			"always",
			{
				"except": ["first-nested", "after-comment", "after-declaration"]
			}
		]
	}
}
```

**Comment by jeddy3** [comment]
FYI, there's a [section in the docs](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/about-rules.md#-empty-line-before-and--max-empty-lines-rules) that does into more detail about the `*-empty-line-before` rules.

**Comment by jeddy3** [comment]
@sindresorhus Does the suggested configuration resolve the issue for you?

**Comment by jeddy3** [comment]
Triaging issues... and closing for now. Feel free to reopen if the comments above don't answer your request satisfactorily.
